### ISSUE TITLE
The windows desktop app is not recognizing the default branch in the repo

### ISSUE THREAD

**Issue by dsbt01** [desktop/desktop#14266]
### Search existing issues

- [X] I have searched the issue tracker and confirmed that I was unable to find existing issues for the problem I am experiencing.

### The problem

I have changed my default branch in the repo to Debug since that is the branch I want any new branches to be used as the base.  The desktop app keeps showing and using the Master branch as the base.  I had already problems doing changes in the wrong branch because of this.  I already tried to sync and all
![DesktopAppWithWrongDefaultBranch](https://user-images.githubusercontent.com/24510933/161043588-a5c50927-b0f2-412f-882c-cd9922868fe6.png)
![GitDefaultBranch](https://user-images.githubusercontent.com/24510933/161043611-ca15bcdf-40fa-4fba-965e-fd35171a6a33.png)
.

### Release version

2.9.12 (x64)

### Operating system

Win 10

### Steps to reproduce the behavior

1. Create a new Debug branch based on Master
2. Change your default branch on Git to Debug from Master
3. Sync repos
4. Default branch in your desktop app is still wrong

**Comment by sergiou87** [comment]
Thanks for your report! It seems Desktop is only fetching which is the default branch when the repo is about to be cloned.

**Comment by dsbt01** [comment]
I believe that is the issue actually.  This issue is a bit annoying.  I have to remember every time I create a new Feature/bug branch to pick Debug before I do it or I will have to delete and do it again.  Very appreciated for the bug report.

**Comment by steveward** [comment]
There have been previous reports of this in https://github.com/desktop/desktop/issues/13669 and https://github.com/desktop/desktop/issues/11299, but I've closed those out in favor of tracking this more recent issue.

**Comment by carlmarkegang** [comment]
My organisation is experiencing issues with this too. 
Would appreciate if it could be fixed!

**Comment by LizzyRandall** [comment]
My team actually changes default branches every few months because they denote releases. So I've run into this issue many times and I have to keep finding the real default branch to branch off of and catching up branches I work on, etc. I know cloning is supposed to workaround this issue but just wanted to add another vote for the fix. Thank you for your work on GitHub Desktop, it's a great tool.

**Comment by SimonShepherd** [comment]
Adding Our organisation has encountered this issue as well

**Comment by igroglaz** [comment]
Please fix

**Comment by sergiou87** [comment]
Another report in https://github.com/desktop/desktop/issues/14844 mentioning this also happens with non-GH repositories.

**Comment by ImRodry** [comment]
Hey! Just to add up to this issue, I've had something similar happen but I'm not sure if this particular situation has been reported.
My repo's main branch is `main` and I host my project on `heroku`, which has a git remote we use to push commits to it for it to update. I added the heroku remote and cloned the main branch with the name `heroku`. I have set the default remote to the original one on github but for some reason, GitHub Desktop sees the default branch as the `heroku` branch. Here are some screenshots of this issue:
![image](https://user-images.githubusercontent.com/38259440/175529070-21f1091d-c6d1-4c62-aadd-f509efead33f.png)
![image](https://user-images.githubusercontent.com/38259440/175529107-6b8c07dd-ae7d-45fc-b23f-6141f165effb.png)
And since I saw this command being suggested in one of the older issues that were linked, the output of `git symbolic-ref refs/remotes/origin/HEAD` is `refs/remotes/origin/main`

**Comment by thinkyhead** [comment]
Version 3.0.2 (arm64)

I recently changed the Default Branch on both the upstream and origin repos (for MarlinFirmware/Marlin) and GitHub Desktop failed to recognize it. Not only that, but GitHub Desktop started to periodically reload the window, every couple of minutes, just as I am trying to interact with it, which makes me 😠 and 🤬.

So, I made a new clone (and linked to my old working copy as a file:// remote). It was working okay for a day or so, but now it has decided that the old Default Branch is the current Default Branch, and it has returned to the bad behavior of reloading the window just as I am trying to use it.

If you test GitHub Desktop under this kind of scenario I bet you will be able to find and fix the issue quickly. The most mysterious part is why it decided to change itself and start using the old Default Branch, when previously I couldn't even get it to change at all.

**Comment by ImRodry** [comment]
Desktop keeps suggesting me to make a PR from my default branch, can the priority on this issue be increased please?

**Comment by sergiou87** [comment]
Hello! We just released 3.0.6-beta2 which hopefully fixes this issue. Please give it a try downloading the latest beta from https://desktop.github.com/beta and let us know if it helps!

**Comment by thinkyhead** [comment]
> Hello! We just released 3.0.6-beta2 which hopefully fixes this issue.

I just launched this build and it correctly picks up the correct default branch and is no longer exhibiting the problem of periodically reloading the window. Thanks!

**Comment by ocdtrekkie** [comment]
@sergiou87 Any idea when this will filter down to release channel? I just encountered this issue when I started mass-renaming master to main... and I'm heartened to see the issue is already fixed!

**Comment by sergiou87** [comment]
@ocdtrekkie sorry no ETA yet. Hopefully in the upcoming weeks but it's kind of a big change, so it'd be very useful if people used the beta channel to confirm it works and report any errors they find 🙏

**Comment by sergiou87** [comment]
@ocdtrekkie (and anyone else interested in this) the fix just went out with v3.0.6 of the app!

**Comment by juliantap** [comment]
This is still happening for me. 

- I've uninstalled and reinstalled 3.0.6 from scratch.
- I've deleted the repo from GHD and re-added it. 
- I've pushed, pulled, etc. 
- If I delete the branch that displays as `main` from GHD, another non-`main` branch takes the place as `default`. 
- `main` is confirmed as default in GH.
- This does not happen with all my repos. But in the repos where it does happen, it happens consistently. 

![image](https://user-images.githubusercontent.com/83591660/186949194-e79482a0-12ad-4681-8342-bfeae9195059.png)

![image](https://user-images.githubusercontent.com/83591660/186949330-34be7ec2-6dc5-4ce7-9618-4039a611c981.png)

**Comment by steveward** [comment]
@juliantap I just tested this out  again and confirmed the default branch got updated correctly in GitHub Desktop, so it seems like you may be hitting an edge case. You mentioned deleting `main` locally, and I'm wondering if that caused issues. Could you run `git remote -v show origin` from the command line to see what that outputs for one of the repositories where you are having issues?

**Comment by juliantap** [comment]
Hi @steveward -- thanks for your reply. `HEAD branch:` is `main` for all my repos (the ones that reflect the default branch correctly and the ones that don't). Also, to clarify, I didn't delete `main` locally. I deleted and re-cloned the wonky repos from GHD.

![image](https://user-images.githubusercontent.com/83591660/187530158-56acaa03-e31f-41c6-a674-2ccb924fa74a.png)

**Comment by MichaIng** [comment]
I just faced this with GHD 3.3.14 on Windows 11. I changed the default branch on GitHub and deleted the previous default branch there. GHD however still keeps the old default branch, refreshing, switching branches, restarting GHD, removing and re-adding the repo (keeping files locally) does not help. Deleting the default branch locally is not possible (greyed out) as long as it is the default branch.

Not sure whether it makes a difference that this is a fork used (configured to) contribute to upstream? The old default branch, still used in GHD, matches the upstream default branch.

I have no local `git`, so cannot run the CLI to check Git config. I'll delete and re-clone, which should solve it.

EDIT: Indeed, the fork behaviour does the difference: When changing to "for my own purpose", the default branch switches immediately to the origin default. When switching back to "to contribute to the parent repository", the default switches back to the upstream default. When I deleted the old default branch, it even switches to `upstream/main` (`main` being the previous/upstream default branch), despite/indicating that this branch does not even exist locally. Not sure whether this is intended, but I find it counter-intuitive/weird. IMO, GHD default branch should match the origin, not the upstream.
